starlite/middleware/session.py

```diff
@@ -0,0 +1,233 @@
+
+    Notes:
+        - If a session cookie exceeds 4KB in size it is split. In this case the key will be of the format
+            'session-{segment number}'.
+    """
+    max_age: conint(ge=1) = ONE_DAY_IN_SECONDS * 14  # type: ignore[valid-type]
+    """Maximal age of the cookie before its invalidated."""
+    scopes: conlist(Literal["http", "websocket"], min_items=1, max_items=2) = ["http", "websocket"]  # type: ignore[valid-type]
+    """Scopes for the middleware - options are 'http' and 'websocket' with the default being both"""
+    path: str = "/"
+    """Path fragment that must exist in the request url for the cookie to be valid. Defaults to '/'."""
+    domain: Optional[str] = None
+    """Domain for which the cookie is valid."""
+    secure: Optional[bool] = None
+    """Https is required for the cookie."""
+    httponly: Optional[bool] = True
+    """Forbids javascript to access the cookie via 'Document.cookie'."""
+    samesite: Literal["lax", "strict", "none"] = "lax"
+    """Controls whether or not a cookie is sent with cross-site requests. Defaults to 'lax'."""
+
+    @validator("secret", always=True)
+    def validate_secret(cls, value: SecretBytes) -> SecretBytes:  # pylint: disable=no-self-argument
+        """Ensures that the 'secret' value is 128, 192 or 256 bits.
+
+        Args:
+            value: A bytes string.
+
+        Raises:
+            ValueError: if the bytes string is of incorrect length.
+
+        Returns:
+            A bytes string.
+        """
+        if len(value.get_secret_value()) not in [16, 24, 32]:
+            raise ValueError("secret length must be 16 (128 bit), 24 (192 bit) or 32 (256 bit)")
+        return value
+
+
+class SessionMiddleware(MiddlewareProtocol):
+    def __init__(
+        self,
+        app: "ASGIApp",
+        config: SessionCookieConfig,
+    ):
+        """Starlite SessionMiddleware.
+
+        Args:
+            app: The 'next' ASGI app to call.
+            config: SessionCookieConfig instance.
+        """
+        super().__init__(app)
+        self.app = app
+        self.config = config
+        self.aesgcm = AESGCM(config.secret.get_secret_value())
+
+    def dump_data(self, data: Any) -> List[bytes]:
+        """Given orjson serializable data, including pydantic models and numpy
+        types, dump it into a bytes string, encrypt, encode and split it into
+        chunks of the desirable size.
+
+        Args:
+            data: Data to serialize, encrypt, encode and chunk.
+
+        Notes:
+            - The returned list is composed of a chunks of a single base64 encoded
+                string that is encrypted using AES-CGM.
+
+        Returns:
+            List of encoded bytes string of a maximum length equal to the 'CHUNK_SIZE' constant.
+        """
+        serialized = dumps(data, default=Response.serializer, option=OPT_SERIALIZE_NUMPY)
+        aad = dumps({"time_to_live": round(time.time()) + self.config.max_age})
```

Comment:
`time_to_live` --> `ttl`, this is an accepted abbreviation. 

```suggestion
        aad = dumps({"ttl": round(time.time()) + self.config.max_age})
```

This though is not the correct term - because the `ttl` is actually the `max_age`. The term should be `expiration` or `expiry`.

Reply:
How about `expires_at`?

Reply:
also works.